### PR TITLE
Switch to new SentrySwift library

### PR DESCRIPTION
**Pull request by edenman**
@jschmid lemme know what you think (in particular, not sure if you'd want to make the version# actually 3.0 for this big of a change)

### CODE REVIEW

File: Corvus.m [hunk 1]
```diff
@@ -21,46 +21,71 @@ + (Corvus *)sharedInstance {
   return sharedInstance;
 }
 
+- (Stacktrace *)buildStacktrace:(DDLogMessage *)logMessage {
+  // NSArray<NSString *> *symbols = [NSThread callStackSymbols];
+  // TODO parse the actual symbols[c] string, if we're able to get the real ones
```
Comment (jschmid) [inline Corvus.m:26]: Can you please tell me what are those TODOs?

File: Corvus.podspec [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Pod::Spec.new do |s|
   s.name         = "Corvus"
-  s.version      = "2.0.0"
-  s.summary      = "Log to Sentry/Raven using CocoaLumberjack"
+  s.version      = "2.1.0"
```
Comment (jschmid) [inline Corvus.podspec]: As you suggested, I would prefer to use v3.0.0

**Review by jschmid — CHANGES_REQUESTED** [review]
Thanks for the PR!
I added a few comments.

File: Corvus.m (hunk 1 above)
Comment (edenman) [reply Corvus.m:26]: In my opinion, it'd be great to send up the full stacktrace to Sentry (only for `DDLogError`, probably?).  Unfortunately, the Corvus logger is called by CocoaLumberjack on a worker thread, where the original call stack is no longer available.  
- I've got a question out to the CocoaLumberjack folks [here](https://github.com/CocoaLumberjack/CocoaLumberjack/issues/276) about this.
- Happy to remove the TODOs for now if you'd prefer, just lmk

Comment (jschmid) [reply Corvus.m:26]: Got it! 
Let's keep the todo then :+1:

**Review by jschmid — APPROVED** [review]
Thanks! I'll do a release tonight.

**Comment by jschmid** [comment]
@edenman I can't release it to CocoaPods Trunk due to https://github.com/CocoaPods/CocoaPods/issues/5661 but I'll do when it's fixed!

**Comment by edenman** [comment]
@jschmid thanks!

**Comment by edenman** [comment]
@jschmid can you make this release please?  I'd like to finally switch over to Sentry-Swift so we can start taking advantage of https://docs.sentry.io/clients/cocoa/advanced/#change-event-before-sending-it

**Comment by jschmid** [comment]
@edenman sorry for the delay. I'm not using this lib currently and forgot about the update.

Can you please try v3.1.4 ?

**Comment by edenman** [comment]
Hm.  When I upgrade, my app crashes every time.  Not sure how this was working before :/  Logs:
```
2016-12-16 15:34:22.139437 recharge-consumer-ios[68766:2423485] [] __nwlog_err_simulate_crash_libsystem libsystem simulate crash unavailable "libsystem_network.dylib: netcore_create_control_socket :: socket(PF_SYSTEM, SOCK_DGRAM, SYSPROTO_CONTROL) failed: [24] Too many open files"
2016-12-16 15:34:22.144255 recharge-consumer-ios[68766:2423485] [] netcore_create_control_socket socket(PF_SYSTEM, SOCK_DGRAM, SYSPROTO_CONTROL) failed: [24] Too many open files, dumping backtrace:
        [x86_64] libnetcore-856.20.4
    0   libsystem_network.dylib             0x0000000114339682 __nw_create_backtrace_string + 123
    1   libsystem_network.dylib             0x000000011432d1a8 netcore_create_control_socket + 411
    2   libsystem_network.dylib             0x000000011434ff09 nw_get_host_stats + 62
    3   libnetwork.dylib                    0x00000001162f378b nw_endpoint_resolver_start_next_child + 1382
    4   libdispatch.dylib                   0x00000001140e50cd _dispatch_client_callout + 8
    5   libdispatch.dylib                   0x00000001140c11c3 _dispatch_continuation_pop + 1091
    6   libdispatch.dylib                   0x00000001140d61e2 _dispatch_source_latch_and_call + 195
    7   libdispatch.dylib                   0x00000001140ced7d _dispatch_source_invoke + 1098
    8   libdispatch.dylib                   0x00000001140c312c _dispatch_queue_serial_drain + 941
    9   libdispatch.dylib                   0x00000001140c3b9f _dispatch_queue_invoke + 1073
    10  libdispatch.dylib                   0x00000001140c63b7 _dispatch_root_queue_drain + 720
    11  libdispatch.dylib                   0x00000001140c608b _dispatch_worker_thread3 + 123
    12  libsystem_pthread.dylib             0x000000011448f736 _pthread_wqthread + 1299
    13  libsystem_pthread.dylib             0x000000011448f211 start_wqthread + 13
2016-12-16 15:34:22.144536 recharge-consumer-ios[68766:2423485] [] __nwlog_err_simulate_crash_libsystem libsystem simulate crash unavailable "libsystem_network.dylib: nw_get_host_stats :: netcore_create_control_socket failed"
2016-12-16 15:34:22.144867 recharge-consumer-ios[68766:2423485] [] nw_get_host_stats netcore_create_control_socket failed, dumping backtrace:
        [x86_64] libnetcore-856.20.4
    0   libsystem_network.dylib             0x0000000114339682 __nw_create_backtrace_string + 123
    1   libsystem_network.dylib             0x000000011434ffa6 nw_get_host_stats + 219
    2   libnetwork.dylib                    0x00000001162f378b nw_endpoint_resolver_start_next_child + 1382
    3   libdispatch.dylib                   0x00000001140e50cd _dispatch_client_callout + 8
    4   libdispatch.dylib                   0x00000001140c11c3 _dispatch_continuation_pop + 1091
    5   libdispatch.dylib                   0x00000001140d61e2 _dispatch_source_latch_and_call + 195
    6   libdispatch.dylib                   0x00000001140ced7d _dispatch_source_invoke + 1098
    7   libdispatch.dylib                   0x00000001140c312c _dispatch_queue_serial_drain + 941
    8   libdispatch.dylib                   0x00000001140c3b9f _dispatch_queue_invoke + 1073
    9   libdispatch.dylib                   0x00000001140c63b7 _dispatch_root_queue_drain + 720
    10  libdispatch.dylib                   0x00000001140c608b _dispatch_worker_thread3 + 123
    11  libsystem_pthread.dylib             0x000000011448f736 _pthread_wqthread + 1299
    12  libsystem_pthread.dylib             0x000000011448f211 start_wqthread + 13
2016-12-16 15:34:22.146119 recharge-consumer-ios[68766:2423485] [] __nwlog_err_simulate_crash_libsystem libsystem simulate crash unavailable "libsystem_network.dylib: nw_route_get_ifindex :: socket(PF_ROUTE, SOCK_RAW, PF_ROUTE) failed: [24] Too many open files"
2016-12-16 15:34:22.146524 recharge-consumer-ios[68766:2423485] [] nw_route_get_ifindex socket(PF_ROUTE, SOCK_RAW, PF_ROUTE) failed: [24] Too many open files, dumping backtrace:
        [x86_64] libnetcore-856.20.4
    0   libsystem_network.dylib             0x0000000114339682 __nw_create_backtrace_string + 123
    1   libsystem_network.dylib             0x000000011432e7d1 nw_route_get_ifindex + 774
    2   libsystem_network.dylib             0x000000011434612b nw_path_simulator_get_path_result + 145
    3   libsystem_network.dylib             0x0000000114345eb0 nw_path_create_evaluator_for_endpoint + 289
    4   libnetwork.dylib                    0x00000001162dbde1 nw_endpoint_handler_start + 415
    5   libnetwork.dylib                    0x00000001162f3ae5 nw_endpoint_resolver_start_next_child + 2240
    6   libdispatch.dylib                   0x00000001140e50cd _dispatch_client_callout + 8
    7   libdispatch.dylib                   0x00000001140c11c3 _dispatch_continuation_pop + 1091
    8   libdispatch.dylib                   0x00000001140d61e2 _dispatch_source_latch_and_call + 195
    9   libdispatch.dylib                   0x00000001140ced7d _dispatch_source_invoke + 1098
    10  libdispatch.dylib                   0x00000001140c312c _dispatch_queue_serial_drain + 941
    11  libdispatch.dylib                   0x00000001140c3b9f _dispatch_queue_invoke + 1073
    12  libdispatch.dylib                   0x00000001140c63b7 _dispatch_root_queue_drain + 720
    13  libdispatch.dylib                   0x00000001140c608b _dispatch_worker_thread3 + 123
    14  libsystem_pthread.dylib             0x000000011448f736 _pthread_wqthread + 1299
    15  libsystem_pthread.dylib             0x000000011448f211 start_wqthread + 13
...
```
(the logs are flooded with hundreds of these stacktraces and the app eventually crashes)

SentrySwift 1.3.0
Swift 2.3
XCode 8.1 beta (8T47)
iPhone SE Simulator running 10.1
CocoaLumberjack 2.4.0 + Corvus 1.3.4

I'm setting up Sentry+Corvus in my AppDelegate's `didFinishLaunchingWithOptions` method thusly:
```
		let sentryClient = SentryClient(dsnString: myDSN)!
		SentryClient.shared = sentryClient
		sentryClient.startCrashHandler()
		DDLog.addLogger(Corvus.sharedInstance(), withLevel: .Warning)
```

Things I've tried:
- removing Corvus entirely: no crash
- switching Corvus back to calling captureMessage (with no stacktrace stuff): still crashes

Maybe it's something about the fact that we're calling SentrySwift from the CocoaLumberjack worker thread?

**Comment by edenman** [comment]
Tried wrapping `captureMessage` with
```
		dispatch_async(dispatch_get_main_queue(), ^{
			[[SentryClient shared] captureMessage:logMsg level:sentryLogLevel];
		});
```
no luck, still crashes :(

**Comment by edenman** [comment]
Added some logging to SentrySwift, looks like i'm getting a bunch of 429 response codes, presumably because I'm sending a ton of log messages to Sentry and it's not batching/throttling them at all 😭

**Comment by edenman** [comment]
Filed as https://github.com/getsentry/sentry-swift/issues/96 bc I'm now pretty certain this isn't a Corvus issue.  Sorry for the noise :)

**Comment by jschmid** [comment]
Thanks for the update! Glad to see that one's work is used by others!

Let me know if it's fixed. Thanks!

**Pull request merged**